api/lib/infrastructure/repositories/campaign-collective-result-repository.js

```diff
@@ -108,24 +79,18 @@ function _computeKERankWithinUserAndSkillByDateDescending(qb) {
   });
 }
 
-function _groupByCompetenceId(targetedSkills) {
-  const targetedSkillsByCompetenceId = _.groupBy(targetedSkills, 'competenceId');
-
-  return { targetedSkillsByCompetenceId };
-}
-
-function _forgeCampaignCompetenceCollectiveResults(campaignId, competences, participantCount, targetedSkillsByCompetenceId, participantsKECountByCompetenceId) {
-  return _(targetedSkillsByCompetenceId).keys().map((competenceId) => {
-    const competence = _.find(competences, { id: competenceId });
-    const averageValidatedSkills = (participantCount && competenceId in participantsKECountByCompetenceId) ?
-      participantsKECountByCompetenceId[competenceId] / participantCount : 0;
+function _forgeCampaignCompetenceCollectiveResults(campaignId, targetProfile, participantCount, participantsKECountByCompetenceId) {
+  return _(targetProfile.competences).map((competence) => {
```

Comment:
Si je ne m'abuse maintenant `targetProfile.competences` est un tableau, on doit donc pouvoir se passer de lodash : 

```suggestion
  return targetProfile.competences.map((competence) => {
```

Reply:
Je pense que c'est notamment pour pouvoir enquiller avec le sortBy qui suit